specs/opcode/05SDIV_07SMOD.md

```diff
@@ -0,0 +1,67 @@
+# SDIV and SMOD opcodes
+
+## Procedure
+
+### EVM behavior
+
+Pop two EVM words `a` and `b` from the stack, and push `c` to the stack. All `a`, `b` and `c` are considered as `signed` values, and `a_abs`, `b_abs` and `c_abs` are absolute values of `a`, `b` and `c`.
+
+- For opcode `SDIV`
+  - `c = 0` when `b == 0`.
+  - `c = -(1 << 255)` when `a == -(1 << 255)` and `b == -1`.
+  - For other cases, compute `c_abs = a_abs // b_abs`,  and `sign(c) = sign(a // b)`.
+- For opcode `SMOD`
+  - `c = 0` when `b == 0`.
+  - For other cases, compute `c_abs = a_abs % b_abs` and `sign(c) = sign(a)`.
+
+### Circuit behavior
+
+To prove the `SDIV/SMOD` opcode, we first get the stack word `pop1`, `pop2` and `push`, and get the absolute value `pop1_abs`, `pop2_abs` and `push_abs`.
+Then use these values to calculate `quotient`, `divisor`, `remainder` and `dividend` for opcode `SDIV` and `SMOD` respectively. And get the absolute value `quotient_abs`, `divisor_abs`, `remainder_abs` and `dividend_abs`.
+Then use these four absolute values to construct a `MulAddWordsGadget` (as described in `02MUL_04DIV_06MOD.md`). The constraints should be matched in `MulAddWordsGadget` since we use the absolute values of `quotient`, `divisor`, `remainder` and `dividend`.
+
+Now putting things together, the constraints can be constructed as follows:
+
+1. `MulAddWordsGadget` overflow:
+
+* It should always be zero.
```

Comment:
tweak the format here. just use tab indent this line. no need to add `*`. and same applies to subsequent bullets.